app/scripts/modules/core/pipeline/config/stages/webhook/webhookStage.ts

```diff
@@ -0,0 +1,84 @@
+import {module, IScope, extend} from 'angular';
+
+import {JSON_UTILITY_SERVICE, JsonUtilityService} from 'core/utils/json/json.utility.service';
+
+interface IViewState {
+  waitForCompletion?: boolean;
+  useStatusUrlFromLocationHeader?: boolean;
+}
+
+interface ICommand {
+  errorMessage?: string;
+  invalid?: boolean;
+  payloadJSON: string;
+}
+
+export class WebhookStage {
+  static get $inject() {
+    return ['stage', '$scope', 'jsonUtilityService'];
+  }
+
+  public command: ICommand;
+  public viewState: IViewState;
+  public methods: string[];
+
+  constructor(public stage: any,
+              $scope: IScope,
+              private jsonUtilityService: JsonUtilityService) {
+    this.methods = ['GET', 'POST', 'PUT'];
+
+    this.viewState = {
+      waitForCompletion: this.stage.waitForCompletion || this.stage.waitForCompletion === undefined,
+      useStatusUrlFromLocationHeader: this.stage.useStatusUrlFromLocationHeader || this.stage.useStatusUrlFromLocationHeader === undefined
+    };
+
+    $scope.$watch('stage.payload', () => this.updatePayload());
+
+    this.command = {
+      payloadJSON: this.jsonUtilityService.makeSortedStringFromObject(this.stage.payload),
+    };
+  }
+
+  public updatePayload(): void {
+    this.command.invalid = false;
+    this.command.errorMessage = '';
+    try {
+      const parsed = JSON.parse(this.command.payloadJSON);
+      extend(this.stage.payload, parsed);
+    } catch (e) {
+      this.command.invalid = true;
+      this.command.errorMessage = e.message;
+    }
+  }
+
+  public waitForCompletionChanged(): void {
+    this.stage.waitForCompletion = this.viewState.waitForCompletion;
+  }
+
+  public useStatusUrlFromLocationHeader(): void {
+    this.stage.useStatusUrlFromLocationHeader = this.viewState.useStatusUrlFromLocationHeader;
```

Comment:
Is there any reason for having a separate property on `viewState` that just gets mapped to a value on the stage?